xfields/beam_elements/beambeam3d.py

```diff
@@ -321,12 +384,15 @@ def _allocate_xobject(self, n_slices, **kwargs):
             slices_other_beam_Sigma_34_star=n_slices,
             slices_other_beam_Sigma_44_star=n_slices,
             slices_other_beam_num_particles=n_slices,
+            slices_other_beam_num_macroparticles=n_slices,
```

Comment:
If the slice contains 0 particles, the mean and sigma are 0. If the slice contains 1 particle, the sigma is 0. If the sigma is 0 then the cerrf returns nans, so the field will be nan. Instead of the if on the macroparticles I could add an if on sigma > 0, but there has to be an if anyways.

Reply:
see [4 hours ago](https://github.com/xsuite/xfields/pull/50#discussion_r1003030751)